packages/cadl-ranch-specs/http/overload/main.cadl

```diff
@@ -0,0 +1,74 @@
+import "@cadl-lang/rest";
+import "@azure-tools/cadl-ranch-expect";
+
+using Cadl.Http;
+
+@doc("Overload")
+@supportedBy("dpg")
+@scenarioService("/overload")
+namespace Overload;
+
+@scenario
+@scenarioDoc("""
+send "hello world" with `conten-type` "text/plain";
+send binary data "hello world" with `conten-type` "application/octet-stream"
+""")
+@route("/bytesOrString")
+@post
+op uploadBytesOrString(
+  @body data: string | bytes,
+  @header contentType: "text/plain" | "application/octet-stream"
+): void;
+
+@scenario
+@scenarioDoc("""
+TODO: optimize cadl-ranch to skip it because the op is only for overload and no need to test
+""")
+@overload(uploadBytesOrString)
+@route("/bytesOrString/skip1")
+@post
+op uploadBytesOrStringSkip1(@body data: string, @header contentType: "text/plain"): void;
+
+@scenario
+@scenarioDoc("""
+TODO: optimize cadl-ranch to skip it because the op is only for overload and no need to test
+""")
+@overload(uploadBytesOrString)
+@route("/bytesOrString/skip2")
+@post
+op uploadBytesOrStringSkip2(@body data: bytes, @header contentType: "application/octet-stream"): void;
+
+model Data {
+  hello: string;
+}
+
+@scenario
+@scenarioDoc("""
+send "hello world" with `conten-type` "text/plain";
+send
+```json
+{ "hello": "world"}
+```
+with `conten-type` "application//json"
+""")
+@route("/JsonOrString")
+@post
+op uploadJsonOrString(@body data: string | Data, @header contentType: "text/plain" | "application/json"): void;
+
+@scenario
+@scenarioDoc("""
+TODO: optimize cadl-ranch to skip it because the op is only for overload and no need to test
+""")
+@overload(uploadJsonOrString)
+@route("/JsonOrString/skip3")
+@post
+op uploadJsonOrStringSkip3(@body data: string, @header contentType: "text/plain"): void;
```

Comment:
what does skip3 mean?

Reply:
just means "I have to define it to avoid compiler error but I don't want to generate API for it"

Reply:
Is it just the overload to `uploadJsonOrString`, which means its name is irrelavent?

Reply:
Why do we need to update the route? I think in service side, there is only one route for this.

Reply:
> Is it just an overload to `uploadJsonOrString`, which means its name is irrelevant?

We have to make a different name for overload op otherwise cadl compiler will report error

Reply:
> Why do we need to update the route? I think in service side, there is only one route for this.

It is the limit of cadl-ranch. Without the `rout`, cadl-ranch will report error

Reply:
But there is only one rest api which will accept multiple content-type. 
If you want to use @overload to split to different sdk methods for multi content-type, the request Uri/route should be the same.

Reply:
> Without the rout, cadl-ranch will report error

Then we should fix cadl-ranch instead of putting this workaround.

Reply:
this is a cadl issue with route right now, we're blocked on cadl for this